WIP - HOFD-71: govuk design update

Rhodine-orleans-lindsay opened the pull request:
## What?
-  Apply GovUK design update to the demo app 
## Why?
- To bring it in line with GovUK design system guidelines
## How?
- update hof package to govuk design update
- amend field.js, fields.json and pages.json to follow govuk design system guidelines on dates, radio and checkbox format
- add maincontent-left view
- add new line to or.html and guidance-form-link.html
- amend confirmation, weapons and layout page views
- amend anchor tag css to follow govuk design system guidelines
- remove unnecessary css
- amend steps.js for acceptance tests.
-  amend landing page content to reflect removal of navbar at top of page
- amend complaint details textarea field configs to match govuk design recommendations
- amend confirm page to match govuk design recommendations
- update to node 18
- amend ingress annotation to build image successfully in drone

##Testing
- Tested locally

Rhodine-orleans-lindsay committed:
HOFD-71: govuk design update

- update hof package to govuk design update
- amend field.js, fields.json and pages.json to follow govuk design system guidelines on dates, radio and checkbox format
- add new line to or.html and guidance-form-link.html
- amend confirmation, weapons and layout page views
- amend anchor tag css to follow govuk design system guidelines
- remove unnecessary css
- add dev script to package.json
- change nav bar to page with radio fields.
- Format title for start and form-guidance pages. Format position of form-guidance page
- add classnames to town, postcode and phone fields to format size
- fix drone build issue
- amend tests to reflect choose a form page

sulthan-ahmed commented on apps/demo/fields.js:
```diff
@@ -113,7 +112,10 @@ module.exports = {
-    ]
+    ],
+    legend: {
+      className: 'govuk-fieldset__legend--m'
```
this interesting, what does this do?

sulthan-ahmed replied:
great stuff, really good idea to add 🙂

sulthan-ahmed commented on apps/demo/views/confirmation.html:
```diff
@@ -1,15 +1,7 @@
+     {{>partials-confirmation-alert}}
+     <a href="/" class="button" role="button">{{#t}}buttons.start-again{{/t}}</a>
+   {{/page-content}}
+ {{/partials-page}}
```
👍

sulthan-ahmed commented on apps/demo/views/form-guidance-link.html:
```diff
@@ -5,4 +5,4 @@
   {{/page-content}}
-{{/partials-page}}
\ No newline at end of file
+{{/partials-page}}
```
👍

sulthan-ahmed commented on apps/demo/views/layout.html:
```diff
@@ -63,12 +66,12 @@
+        <li class="govuk-footer__inline-list-item"><a  class="govuk-footer__link" href="/terms-and-conditions">{{#t}}base.terms{{/t}}</a></li>
       {{/footerSupportLinks}}
     </ul>
   {{/footerSupportLinks}}
```
is there a reason to override the main layout in hof? I think you can change the footer links dynamically, I think it's done in my pr for end of tenancy

Rhodine-orleans-lindsay replied:
the reason the layout is overriding is because of the nav bar not the footer -  the change I made to the footer links makes them the same as the main layout in hof. We could change the nav bar to it's own page with the options (similar to the sandbox) and then we won't need to override the main layout and can remove that view from the demo app?

sulthan-ahmed replied:
separate nav bar page sounds good 👍

sulthan-ahmed commented on apps/demo/views/partials/maincontent-left.html:
```diff
@@ -0,0 +1,10 @@
+    </header>
+    {{$content}}{{/content}}
+  </div>
+{{/content-outer}}
```
I think @Rhyad-Zergane-HO has fixed this now, if not can you put it into the main bit of hof

Rhodine-orleans-lindsay replied:
I have removed this now as this has been fixed in hof.

sulthan-ahmed commented on package.json:
```diff
@@ -5,11 +5,11 @@
   "scripts": {
     "start": "node server.js",
-    "start:dev": "hof-build watch",
+    "start:dev": "hof-build watch --env",
```
I would just add a script called `dev` with --env
some devs don't use .env files like me and you

sulthan-ahmed reviewed (commented):
looks mostly good but some questions.  Also, I think the hof-demo app should avoid overriding templates because this is the example for other services imho

sulthan-ahmed commented:
love your git commit message

sulthan-ahmed commented on .drone.yml:
```diff
@@ -73,7 +73,7 @@ steps:
     image: 340268328991.dkr.ecr.eu-west-2.amazonaws.com/acp/dind
     commands:
-      - n=0; while [ "$n" -lt 60 ] && [ ! docker stats --no-stream ]; do n=$(( n + 1 )); sleep 1; done
+      - n=0; while [ "$n" -lt 60 ] && ! docker stats --no-stream >/dev/null 2>&1; do n=$(( n + 1 )); sleep 1; done
```
just curious can you explain this to me?

Rhodine-orleans-lindsay replied:
This was ACP's fix for an issue with the dind (Docker in Docker) image for Drone builds. The issue was causing the build_image step to fail in Drone.

sulthan-ahmed commented on apps/demo/index.js:
```diff
@@ -8,9 +8,64 @@ const InternationalPhoneNumber = require('./behaviours/international-number');
+    //     },
+        
+    //   ]
+    // },
```
did you mean to have comment code?

Rhodine-orleans-lindsay replied:
I have removed the comments now

sulthan-ahmed commented on apps/demo/translations/src/en/buttons.json:
```diff
@@ -0,0 +1,3 @@
+{
+  "start": "Start"
+}
```
newline, I tend to review my own pr before I do a PR

also, vscode or any other editor will do newliines for you

Rhodine-orleans-lindsay replied:
I have added a new line now

sulthan-ahmed commented on assets/scss/app.scss:
```diff
@@ -26,3 +26,10 @@
+
+  a {
+    @extend .govuk-link;
+  }
```
saw this in Rob's PR, good idea 👍

sulthan-ahmed reviewed (commented):
nice work, just some suggestions

Rhodine-orleans-lindsay committed:
HOFD-71: govuk design update
- update to latest hof package
- amend landing page content to reflect removal of navbar at top of page
- amend complaint details textarea field configs to match govuk design recommendations
- amend confirm page to match govuk design recommendations
- update to node 18

Rhodine-orleans-lindsay committed:
amend ingress annotation

ChisomOguibe11 reviewed (approved):
lgtm

Rhodine-orleans-lindsay committed:
amend node versions and ingress file
